buildSrc/src/main/kotlin/mockito.root.releasing-conventions.gradle.kts

```diff
@@ -0,0 +1,73 @@
+import gradle.kotlin.dsl.accessors._edf0c2271024fd9311959755ab251c7b.ext
+
+plugins {
+    id("org.shipkit.shipkit-auto-version")
+    id("org.shipkit.shipkit-changelog")
+    id("org.shipkit.shipkit-github-release")
+
+    id("io.github.gradle-nexus.publish-plugin")
+}
+
+val isSnapshot = version.toString().endsWith("SNAPSHOT")
+tasks {
+    generateChangelog {
+        githubToken = providers.environmentVariable("GITHUB_TOKEN").get()
+        previousRevision = project.ext["shipkit-auto-version.previous-tag"].toString()
+        repository = "mockito/mockito"
+    }
+
+    githubRelease {
+        enabled = !isSnapshot
+        dependsOn(generateChangelog)
+        githubToken = providers.environmentVariable("GITHUB_TOKEN").get()
+        newTagRevision = providers.environmentVariable("GITHUB_SHA").get()
+        repository = generateChangelog.get().repository
```

Comment:
WDYT about defining constants values instead of referencing separate tasks here. That removes the need for eager configuration

Reply:
Indeed that came to mind. I avoided this here to make a 1:1 translation. But this would be better in my opinion.